### PR TITLE
[Editor] Update SceneUniqueName button text for clarity

### PR DESCRIPTION
**Pull request by yahkr**
- Update on hover message for clarity. 
- Remove "Click to disable this." when the button is disabled.

Enabled
image
Disabled
image

### CODE REVIEW

**Comment by timothyqiu** [comment]
I think the same button showing different hint text depending on the context might be unexpected by many users. It could be harmonized into more precise language, for example:

> This node can be accessed from within anywhere in its owning scene by preceding it with the '%s' prefix in a node path.

**Comment by yahkr** [comment]
> I think the same button showing different hint text depending on the context might be unexpected by many users. It could be harmonized into more precise language, for example:
> 
> > This node can be accessed from within anywhere in its owning scene by preceding it with the '%s' prefix in a node path.

I do like this combined language, however I fear users may not understand what ownership really is, but maybe that's not the problem to try and solve.

Thoughts on?
> This node can be accessed from anywhere within the scene it belongs to by using the '%s' prefix in the node path.

I would also just be happy if the message was the same and we at least removed the "Click to disable this" as it's not immediately clear why you can't click it.

**Comment by AdriaandeJongh** [comment]
I think the tooltip is a great place to mention what pressing the button does, because it isn’t really explained anywhere else what it does.

File: editor/gui/scene_tree_editor.cpp [hunk 1]
```diff
@@ -344,7 +344,9 @@ void SceneTreeEditor::_add_nodes(Node *p_node, TreeItem *p_parent) {
 		}
 
 		if (p_node->is_unique_name_in_owner()) {
-			item->add_button(0, get_editor_theme_icon(SNAME("SceneUniqueName")), BUTTON_UNIQUE, p_node->get_owner() != EditorNode::get_singleton()->get_edited_scene(), vformat(TTR("This node can be accessed from within anywhere in the scene by preceding it with the '%s' prefix in a node path.\nClick to disable this."), UNIQUE_NODE_PREFIX));
+			bool is_enabled = p_node->get_owner() == EditorNode::get_singleton()->get_edited_scene();
+			String button_text = vformat(TTR("This node can be accessed from anywhere within the scene it belongs to by using the '%s' prefix in the node path%s."), UNIQUE_NODE_PREFIX, is_enabled ? "\nClick to disable this." : "");
```
Comment (AThousandShips) [inline editor/gui/scene_tree_editor.cpp:348]:
```suggestion
			String button_text = vformat(TTR("This node can be accessed from anywhere within the scene it belongs to by using the '%s' prefix in the node path.%s"), UNIQUE_NODE_PREFIX, is_enabled ? ("\n" + TTR("Click to disable this.")) : "");
```
The additional text needs to be translated, and the period should still be at the end of the first sentence

File: editor/gui/scene_tree_editor.cpp [hunk 2]
```diff
@@ -344,7 +344,9 @@ void SceneTreeEditor::_add_nodes(Node *p_node, TreeItem *p_parent) {
 		}
 
 		if (p_node->is_unique_name_in_owner()) {
-			item->add_button(0, get_editor_theme_icon(SNAME("SceneUniqueName")), BUTTON_UNIQUE, p_node->get_owner() != EditorNode::get_singleton()->get_edited_scene(), vformat(TTR("This node can be accessed from within anywhere in the scene by preceding it with the '%s' prefix in a node path.\nClick to disable this."), UNIQUE_NODE_PREFIX));
+			bool is_disabled = p_node->get_owner() != EditorNode::get_singleton()->get_edited_scene();
+			String button_text = vformat(TTR("This node can be accessed from anywhere within the scene it belongs to by using the '%s' prefix in the node path.%s"), UNIQUE_NODE_PREFIX, is_disabled ? "" : "\n" + TTR("Click to disable this."));
```
Comment (Mickeon) [inline editor/gui/scene_tree_editor.cpp:348]: This is a very unorthodox way of doing this .
```suggestion
			String button_text = vformat(TTR("This node can be accessed from anywhere within the scene it belongs to by using the '%s' prefix in the node path."), UNIQUE_NODE_PREFIX)) + (is_disabled ? "" : "\n" + TTR("Click to disable this."));
```

**Review by Mickeon — APPROVED** [review]
This is fine, actually.

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**